dapp/src/components/buySell/SellWidget.js

```diff
@@ -221,10 +222,46 @@ const SellWidget = ({
         result = await vaultContract.redeemAll(minStableCoinsReceivedBN, {
           gasLimit,
         })
-        storeTransaction(result, `redeem`, returnedCoins, coinData)
+        receipt = await rpcProvider.waitForTransaction(result.hash)
+        const data = {
+          usdt: 0,
+          dai: 0,
+          usdc: 0,
+          ousd: 0,
+        }
+        await Promise.all(
+          receipt.logs.map(async (log) => {
+            if (
```

Comment:
This chunk of code seems to repeat itself 3 times. Perhaps it would make sense to move this code inside the `storeTransaction` function